miniapp/src/main/java/com/rakuten/tech/mobile/miniapp/api/ApiClient.kt

```diff
@@ -101,13 +106,36 @@ internal class RetrofitRequestExecutor(
     private val retrofit: Retrofit
 ) {
 
-    private inline fun <reified T : ErrorResponse> createErrorConvertor(retrofit: Retrofit) =
+    private var retryCount = 0
+
+    private inline fun <reified T : ErrorResponse> createErrorConverter(retrofit: Retrofit) =
         retrofit.responseBodyConverter<T>(T::class.java, arrayOfNulls<Annotation>(0))
 
-    @Suppress("TooGenericExceptionCaught", "ThrowsCount")
+    @Suppress(
+        "TooGenericExceptionCaught", "ThrowsCount", "MagicNumber",
+        "LongMethod", "NestedBlockDepth", "ComplexMethod"
+    )
     suspend fun <T> executeRequest(call: Call<T>): T = try {
         val response = call.execute()
 
+        // retry network request when there is 500 error code from the server
+        if (response.code() >= 500) {
```

Comment:
I think this should be `response.code() > 500` in order to match the iOS implementation.

Reply:
Oh, I thought, I am already matching it with iOS according to [statusCode >= 500](https://github.com/rakutentech/ios-miniapp/blob/b3bedb0d71d4c6bb2540b9f7968bf7c153dab6fe/MiniApp/Classes/core/MiniAppClient.swift#L99).

Reply:
@khairul-alam-licon Oops, please ignore this comment. Sorry about that.

I didn't look closely at this line and I was thinking it was the `retryCount` line.

Reply:
Oh, no problem 👍 
Yeah, I am looking forward to address other feedbacks accordingly.